elastic/kibana
# [Connector] Adding internal route for requesting ad-hoc ServiceNow access token

Towards https://github.com/elastic/kibana/issues/130306

## Summary

Addresses item 6 from https://github.com/elastic/kibana/issues/130306: `Expose new actions API HTTP route, which allow to fetch an access token for not saved connectors.`

This PR adds an internal route to request an ad-hoc access token from a ServiceNow instance without requiring a saved connector. This allows the Create ServiceNow connector UI to request information from the `api/x_elas2_inc_int/elastic_api/health` endpoint during the ServiceNow connector creation process.

## To Verify
1. Run this PR and submit the following POST request:

```
POST https://localhost:5601/internal/actions/connector/_servicenow_access_token
{
  "apiUrl": "https://ven04334.service-now.com/",
  "config": {
    "clientId": <clientId>,
    "userIdentifierValue": "elastic_integration",
    "jwtKeyId": <jwtKeyId>
  },
  "secrets": {
    "clientSecret": <clientSecret>,
    "privateKey": <privateKey>
  }
}
```

2. Verify you get an access token in response in the form: `Bearer <accessToken>`


### Checklist
